silo/tests/test_readview.py

```diff
@@ -0,0 +1,157 @@
+from django.test import TestCase
+
+from rest_framework.test import APIRequestFactory
+
+import factories
+from silo.api import ReadViewSet
+from silo.models import Read
+
+
+class ReadListViewTest(TestCase):
+    def setUp(self):
+        factories.Organization(id=1)
+        factories.Read.create_batch(4)
+        self.factory = APIRequestFactory()
+        self.user = factories.User(first_name='Homer', last_name='Simpson')
+        self.tola_user = factories.TolaUser(user=self.user)
+
+    def test_list_read_superuser(self):
+        request = self.factory.get('/api/read/')
+        request.user = factories.User.build(is_superuser=True,
+                                            is_staff=True)
+        view = ReadViewSet.as_view({'get': 'list'})
+        response = view(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 4)
+
+    def test_list_read_normal_user(self):
+        request = self.factory.get('/api/read/')
+        request.user = self.tola_user.user
+        view = ReadViewSet.as_view({'get': 'list'})
+        response = view(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 0)
+
+        factories.Read(owner=self.tola_user.user)
+        response = view(request)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.data), 1)
+
+
+class ReadRetrieveViewTest(TestCase):
+    def setUp(self):
+        factories.Organization(id=1)
+        self.user = factories.User(first_name='Homer', last_name='Simpson')
+        self.tola_user = factories.TolaUser(user=self.user)
+        self.read = factories.Read(read_name="test_data",
+                                   owner=self.tola_user.user)
+        self.silo = factories.Silo(owner=self.tola_user.user,
+                                   reads=[self.read])
+        self.factory = APIRequestFactory()
+
+    def test_retrieve_read_superuser(self):
+        request = self.factory.get('/api/read/')
+        request.user = factories.User.build(is_superuser=True,
+                                            is_staff=True)
+        view = ReadViewSet.as_view({'get': 'retrieve'})
+        response = view(request, pk=self.read.id)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['read_name'], self.read.read_name)
+
+    def test_retrieve_read_normal_user(self):
+        request = self.factory.get('/api/read/')
+        request.user = factories.User.build(is_superuser=False,
+                                            is_staff=False)
+        view = ReadViewSet.as_view({'get': 'retrieve'})
+        response = view(request, pk=self.read.id)
+        self.assertEqual(response.status_code, 404)
+
+        request = self.factory.get('/api/read/')
+        read = factories.Read(read_name='My Read', owner=self.tola_user.user)
```

Comment:
You did already create a Read in the `setUp` for this user, use it instead of creating a new one.